src/amberscript/parser.cc

```diff
@@ -1235,5 +1237,66 @@ Result Parser::ParseCopy() {
   return ValidateEndOfStatement("COPY command");
 }
 
+Result Parser::ParseClearColor() {
+  auto token = tokenizer_->NextToken();
+  if (!token->IsString())
+    return Result("missing pipeline name for CLEAR_COLOR command");
+
+  auto* pipeline = script_->GetPipeline(token->AsString());
+  if (!pipeline) {
+    return Result("unknown pipeline for CLEAR_COLOR command: " +
+                  token->AsString());
+  }
+  if (!pipeline->IsGraphics()) {
+    return Result(
+        "CLEAR_COLOR command requires graphics pipeline, got compute");
```

Comment:
This conditional + message make the assumption that a pipeline is either graphics or compute, which may not be future proof? That is nit-picking, but maybe the message should just say "CLEAR_COLOR command argument is not a graphics pipeline" ?

Reply:
Good point. Fixed other instances as well.